src/pretix/plugins/stripe/views.py

```diff
@@ -150,22 +150,28 @@ def webhook(request, *args, **kwargs):
 
     if event_json['data']['object']['object'] == "charge":
         func = charge_webhook
-        objid = event_json['data']['object']['id']
     elif event_json['data']['object']['object'] == "dispute":
         func = charge_webhook
-        objid = event_json['data']['object']['charge']
     elif event_json['data']['object']['object'] == "source":
         func = source_webhook
-        objid = event_json['data']['object']['id']
     else:
         return HttpResponse("Not interested in this data type", status=200)
 
+    if 'payment_intent' in event_json['data']['object'] and event_json['data']['object']['payment_intent']:
+        objid = event_json['data']['object']['payment_intent']
+    elif 'payment_method' in event_json['data']['object'] and event_json['data']['object']['payment_method']:
+        objid = event_json['data']['object']['payment_method']
+    else:
+        objid = event_json['data']['object']['id']
```

Comment:
Doesn't this break the ``event_json['data']['object']['object'] == "dispute"`` case where we actually want the charge ID, not the dispute ID?